### PR TITLE
Monorepo: add VaultPress plugin

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

Instead of working on the VaultPress plugin in its own repo, let's maintain it here, in the monorepo.

This PR brings everything from https://github.com/Automattic/vaultpress/
From now on, changes made here in the monorepo should get pushed to that same repo, that will become the production repo.

This PR:
- adds the plugin base.
- adds most of the plugin files to the phpcs exclusion list.
- adds the plugin files.
- updates the plugin's changelog to match the new format.
- updates the plugin's dependencies to match the format we use here in the monorepo.
- adds the necessary attributes so we can release the plugin from here.
- adds a changelog entry.

#### Jetpack product discussion

* Internal reference: pIFnH-6EL-p2

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* In your local Docker environment, build the VaultPress plugin: `jetpack build plugins/vaultpress`
* When going to the Plugins menu, you should now be able to activate the VaultPress plugin.
* Once activated, you should see a VaultPress submenu under the Jetpack menu.

* Do the checks below pass?

### CODE REVIEW

**Comment by jeherve** [comment]
@anomiex If we were to want to merge this at some point, how would we go about the linting errors? We're introducing new files with lots of linting errors, and that triggers phpcs-changed. Should we add `phpcs:ignoreFile` comments for each file in this PR, and then remove them in a follow-up PR to work around that problem, or is there a different way we should go about this?

**Comment by anomiex** [comment]
That reminds me, we really should make "Linting / PHP Code Sniffer (changes to excluded files only)" required. Right now you could just ignore it and merge anyway.

And that's what I'd recommend as the solution here: make it not-required, merge, then make it required again right away. Same would go for "Linting / ESLint (changes to excluded files only)", if that were failing too. _Not_ any of the others though, just those two.

Or you could just fix the errors. 😉 But I see there's a ton of them, so I see why you don't want to do that immediately.

File: projects/plugins/vaultpress/license.txt
Comment (anomiex) [inline projects/plugins/vaultpress/license.txt:1]: Should be able to get rid of this here. Since composer.json declares the license as "GPL-2.0-or-later", the mirroring build script will copy it in from .github/licenses/GPL-2.0-or-later.txt.

File: projects/plugins/vaultpress/composer.json
Comment (anomiex) [inline projects/plugins/vaultpress/composer.json:36]: You'll want to do something with these. If nothing else, to do `pnpm run build` because you have some build steps defined in package.json.

**Comment by anomiex** [comment]
I note this just copies the files over, without any history. If we do this for real, shouldn't we import the history too like we did when we imported the Jetpack Beta Plugin?

We might also want to edit .github/files/required-review.yaml to let VaultPress people approve changes in the plugin.

File: projects/plugins/vaultpress/composer.json
Comment (anomiex) [reply projects/plugins/vaultpress/composer.json:36]: BTW, that's why the Build job is failing. It needs the stuff that `pnpm run build-production` would do in order to have `vendor/` set up right for the mirroring.

File: projects/plugins/vaultpress/license.txt
Comment (jeherve) [reply projects/plugins/vaultpress/license.txt]: Done.

File: projects/plugins/vaultpress/composer.json
Comment (jeherve) [reply projects/plugins/vaultpress/composer.json]: Good point. Done.

**Comment by jeherve** [comment]
> I note this just copies the files over, without any history. If we do this for real, shouldn't we import the history too like we did when we imported the Jetpack Beta Plugin?

That's a good point, I had not considered that. I just updated the branch to do just that.

> We might also want to edit .github/files/required-review.yaml to let VaultPress people approve changes in the plugin.

Done in 8a68d092aa53fef551bf11720c44db2c1551a232

File: projects/plugins/vaultpress/package.json
Comment (anomiex) [inline projects/plugins/vaultpress/package.json:24]: I note the package.json doesn't do anything other than define these scripts. We might just move the scripts into composer.json and delete the whole file.

File: projects/plugins/vaultpress/composer.json
Comment (anomiex) [inline projects/plugins/vaultpress/composer.json:22]: I note this project doesn't actually have a `src/` directory. Unless there's plans to add one, we may as well delete this bit.
```suggestion
```

**Review by anomiex — COMMENTED** [review]
I checked the built change that will be pushed to the new mirror repo. It looks sensible to me.

The only inline comment I feel strongly about is on required-review.yaml line 21, whatever we decide there we need to decide intentionally. The rest are trivial or suggestions that I wouldn't complain about if you don't take.

At the risk of restating what you already know, remember that we don't want to use "squash and merge" for this PR. Details at https://jetpackcrew.wordpress.com/2021/01/26/beta-plugin-in-monorepo/. You might also consider manually squashing some of the changes in this PR (after afce3f6) before merging.

File: .github/files/required-review.yaml
Comment (kraftbj) [reply .github/files/required-review.yaml:21]: I'd like Crew to always have approval rights in the monorepo, so I'm going to :y: the suggested change and commit it.

Comment (jeherve) [reply .github/files/required-review.yaml:21]: Ah yes, my bad, I had assumed Jetpack approvers would always be allowed to approve, since they're listed at the end of the file.

File: projects/plugins/vaultpress/package.json
Comment (jeherve) [reply projects/plugins/vaultpress/package.json]: Yeah, that makes sense for a PHP-only plugin. is e90ddd3529437ea92ed990041867ebe054fbd898 what you had in mind? I'm not sure we need `COMPOSER_ROOT_VERSION` in there, given https://github.com/Automattic/jetpack/blob/16f7b31b5c57ae218a20f1b6a1c607ea522c9f26/.github/workflows/build.yml#L12 ?

But `COMPOSER_MIRROR_PATH_REPOS` would need to stay, right?

**Comment by jeherve** [comment]
Thanks for the review!

> remember that we don't want to use "squash and merge" for this PR

Yup, I'll deal with the merge myself, hence the "do not merge" label.

> You might also consider manually squashing some of the changes in this PR

👍 I'll squash those and ask for a final approval once https://github.com/Automattic/jetpack/pull/20098#discussion_r654515076 is settled.

File: .github/files/required-review.yaml
Comment (anomiex) [reply .github/files/required-review.yaml:21]: No problem. In case you haven't seen it, it's documented at https://github.com/automattic/action-required-review#requirements-format.

File: projects/plugins/vaultpress/package.json
Comment (anomiex) [reply projects/plugins/vaultpress/package.json]: The `COMPOSER_MIRROR_PATH_REPOS` needs to stay, yes. It should be ok to leave out the `COMPOSER_ROOT_VERSION`, since hopefully anything actually needing that set is already setting it itself.

File: projects/plugins/vaultpress/composer.json
Comment (anomiex) [inline projects/plugins/vaultpress/composer.json:33]: I wonder if we could leave off the `--ignore-platform-reqs` bit.

**Review by anomiex — APPROVED** [review]

**Comment by jeherve** [comment]
This should now be ready for a final review and approval. I'll then disable squashing, merge that PR, re-enable squashing, and also make the "Linting / PHP Code Sniffer (changes to excluded files only)" check required for the future.

File: projects/plugins/vaultpress/composer.json
Comment (anomiex) [inline projects/plugins/vaultpress/composer.json:62]: I see there's a `VAULTPRESS__VERSION` in vaultpress.php, we may as well flag that for checking and updating by our tools.

And `wp-plugin-slug` will be needed to use `tools/deploy-to-svn.sh` to deploy a release.
```suggestion
		"version-constants": {
			"VAULTPRESS__VERSION": "vaultpress.php"
		},
		"wp-plugin-slug": "vaultpress"
	}
```

File: projects/plugins/vaultpress/to-test.md
Comment (anomiex) [inline projects/plugins/vaultpress/to-test.md]: You could keep this if you want (but probably production-exclude it). In #20137 I'm moving Jetpack's to-test.md into projects/plugins/jetpack and adjusting the build so having this file will do the right thing with the Beta plugin.

Comment (jeherve) [reply projects/plugins/vaultpress/to-test.md]: I think it's fine to get rid of it for now. The VaultPress team hasn't relied on it for their past few releases.

File: projects/plugins/vaultpress/composer.json
Comment (anomiex) [inline projects/plugins/vaultpress/composer.json:61]: Oops.
```suggestion
		"release-branch-prefix": "vaultpress",
```

**Review by anomiex — APPROVED** [review]
Needs a master merge or rebase, but otherwise looks good to me.

**Review by anomiex — APPROVED** [review]

**Comment by anomiex** [comment]
Good to go. Setting "Needs Author Reply" rather than "Ready to Merge" just to avoid confusing anyone who doesn't notice the "DO NOT MERGE" tag.

**Pull request merged**
